modules/porous_flow/src/postprocessors/FVPorousFlowFluidMass.C

```diff
@@ -0,0 +1,126 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVPorousFlowFluidMass.h"
+
+registerMooseObject("PorousFlowApp", FVPorousFlowFluidMass);
+
+InputParameters
+FVPorousFlowFluidMass::validParams()
+{
+  InputParameters params = ElementIntegralPostprocessor::validParams();
+  params.addParam<unsigned int>(
+      "fluid_component",
+      0,
+      "The index corresponding to the fluid component that this Postprocessor acts on");
+  params.addRequiredParam<UserObjectName>(
+      "PorousFlowDictator", "The UserObject that holds the list of PorousFlow variable names.");
+  params.addParam<std::vector<unsigned int>>("phase",
+                                             "The index of the fluid phase that this "
+                                             "Postprocessor is restricted to.  Multiple "
+                                             "indices can be entered");
+  params.addRangeCheckedParam<Real>("saturation_threshold",
+                                    1.0,
+                                    "saturation_threshold >= 0 & saturation_threshold <= 1",
+                                    "The saturation threshold below which the mass is calculated "
+                                    "for a specific phase. Default is 1.0. Note: only one "
+                                    "phase_index can be entered");
+  params.addParam<std::string>(
+      "base_name",
+      "For non-mechanically-coupled systems with no TensorMechanics strain calculators, base_name "
+      "need not be set.  For mechanically-coupled systems, base_name should be the same base_name "
+      "as given to the TensorMechanics object that computes strain, so that this Postprocessor can "
+      "correctly account for changes in mesh volume.  For non-mechanically-coupled systems, "
+      "base_name should not be the base_name of any TensorMechanics strain calculators.");
+  params.set<bool>("use_displaced_mesh") = false;
+  params.suppressParameter<bool>("use_displaced_mesh");
+  params.addClassDescription("Calculates the mass of a fluid component in a region");
+  return params;
+}
+
+FVPorousFlowFluidMass::FVPorousFlowFluidMass(const InputParameters & parameters)
+  : ElementIntegralPostprocessor(parameters),
+    _dictator(getUserObject<PorousFlowDictator>("PorousFlowDictator")),
+    _fluid_component(getParam<unsigned int>("fluid_component")),
+    _phase_index(getParam<std::vector<unsigned int>>("phase")),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _has_total_strain(hasMaterialProperty<RankTwoTensor>(_base_name + "total_strain")),
```

Comment:
this logic seems dangerous

if i mispelled base_name, it wont retrieve a total strain and will change the result.

why would not always retrieve the total strain if a base name is passed?

Reply:
Oh, you might want to use a total strain but if you make a typo in the input file, it will just silently not use one and the user won't know why they are getting an incorrect value. I guess we should error if base_name is provided but doesn't actually exist.

This was a copy and paste from the existing post processor, and I think the same logic is used in the energy postprocessor too. I'll make an issue to fix all of these.

Reply:
I've opened an issue to fix this in the postprocessors not included in this PR

Reply:
I added a check and a paramError when base_name is given by doesn't exist. I think this is probably the correct behaviour.